testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkDropTableCompatibility.java

```diff
@@ -0,0 +1,134 @@
+                {SPARK_QUERY_ENGINE, SPARK_QUERY_ENGINE},
+                {SPARK_QUERY_ENGINE, TRINO_QUERY_ENGINE},
+        };
+    }
+
+    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS}, dataProvider = "tableCleanupEngineConfigurations")
+    public void testCleanupOnDropTable(String tableCreatorQueryEngine, String tableDropperQueryEngine)
+    {
+        String tableName = "test_cleanup_on_drop_table" + randomTableSuffix();
+
+        if (TRINO_QUERY_ENGINE.equals(tableCreatorQueryEngine)) {
+            onTrino().executeQuery("CREATE TABLE " + trinoTableName(tableName) + "(col0 INT, col1 INT)");
+        }
+        else if (SPARK_QUERY_ENGINE.equals(tableCreatorQueryEngine)) {
+            onSpark().executeQuery("CREATE TABLE " + sparkTableName(tableName) + "(col0 INT, col1 INT)");
+        }
+        onTrino().executeQuery("INSERT INTO " + trinoTableName(tableName) + " VALUES (1, 2)");
+
+        String tableDirectory = format("%s/%s", warehouseDirectory, tableName);
+        assertFileExistence(tableDirectory, true, "The table directory exists after creating the table");
+        List<String> dataFilePaths = getDataFilePaths(tableName);
+
+        if (TRINO_QUERY_ENGINE.equals(tableDropperQueryEngine)) {
+            onTrino().executeQuery("DROP TABLE " + trinoTableName(tableName));
+            assertFileExistence(tableDirectory, false, "The table directory should be removed after dropping the table from Trino");
+        }
+        else if (SPARK_QUERY_ENGINE.equals(tableDropperQueryEngine)) {
+            onSpark().executeQuery("DROP TABLE " + sparkTableName(tableName));
+            assertFileExistence(tableDirectory, true, "The table directory should not be removed after dropping the table from Spark");
+        }
+        dataFilePaths.forEach(dataFilePath -> assertFileExistence(dataFilePath, false, format("The data file %s removed after dropping the table from Spark", dataFilePath)));
+    }
+
+    private static String sparkTableName(String tableName)
+    {
+        return format("%s.%s.%s", SPARK_CATALOG, TEST_SCHEMA_NAME, tableName);
+    }
+
+    private static String trinoTableName(String tableName)
+    {
+        return format("%s.%s.%s", TRINO_CATALOG, TEST_SCHEMA_NAME, tableName);
+    }
+
+    private void assertFileExistence(String path, boolean exists, String description)
+    {
+        Assertions.assertThat(hdfsClient.exist(path)).as(description).isEqualTo(exists);
+    }
+
+    private static List<String> getDataFilePaths(String icebergTableName)
+    {
+        List<String> filePaths = onTrino().executeQuery(format("SELECT file_path FROM %s.%s.\"%s$files\"", TRINO_CATALOG, TEST_SCHEMA_NAME, icebergTableName)).column(1);
+        return filePaths.stream().map(TestIcebergSparkDropTableCompatibility::getPath).collect(Collectors.toList());
+    }
+
+    private static String getPath(String uri)
+    {
+        try {
+            return new URI(uri).getPath();
```

Comment:
> path of an uri (discarding scheme and authority) seems wrong

The `path` is then given to the `hdfsClient` which makes use of the scheme and authority behind the scenes.

The `hdfs` bash utility makes use of paths instead of URIs

```
hdfs dfs -ls -R /user/hive/warehouse
```


Not necessarily as an argument, I found two instances of this usage pattern:

- `io.trino.tests.product.TestImpersonation#getTableLocation`
- `io.trino.tests.product.hive.util.TableLocationUtils#getTablePath(java.lang.String, int)`

Reply:
> Not necessarily as an argument, I found two instances of this usage pattern:

Can we try better than that, since now we realized it's not necessarily what we should be doing?

note that `new URI(uri).getPath();`  transformation is simply lossy:

```
hdfs://host/path -> path
s3://bucket/path -> path
```

Reply:
> if we need to strip off a known prefix (eg hdfs://....), this can be done with uri.replaceFirst("^\\Q....")`,

I fail to see at the moment what the appropriate regex would be in such a case. Using a host/ip(&port) regex seems fairly complicated for readability.

Reply:
is the hostname always `hadoop-master` in product tests?